src/components/Spawn/index.tsx

```diff
@@ -108,3 +100,41 @@ export const SectionLabel = styled(Body)`
   margin-top: 22px;
   min-width: 175px;
 `;
+
+interface CardItem {
+  label: string;
+  value: JSX.Element;
+}
+
+const CardField: React.FC<CardItem> = ({ label, value }) => (
+  <FieldContainer>
+    <FieldName>{label}</FieldName>
+    <div>{value}</div>
+  </FieldContainer>
+);
+
+type FieldMap<T> = {
+  [key: string]: (T: T) => JSX.Element;
+};
+
+interface DetailsCardProps {
+  type: MyHost | MyVolume;
+  ["data-cy"]?: string;
+  fieldMaps: FieldMap<MyHost | MyVolume>;
+}
+
+export const DetailsCard: React.FC<DetailsCardProps> = ({
+  type,
+  "data-cy": dataCy,
+  fieldMaps,
```

Comment:
This might be better suited living its own file. Since all this is related to just the `DetailsCard`